File: Source/Client/Game/Objects/Moral.vb
Imports Core
Imports Mirage.Sharp.Asfw

Friend Module Moral
#Region "Database"

    Friend Sub ClearMoral(index As Integer)
        Type.Moral(index) = Nothing

        Type.Moral(index).Name = ""
        GameState.Moral_Loaded(index) = 0
    End Sub

    Sub ClearMorals()
        Dim i As Integer

        ReDim Type.Moral(MAX_MORALS)

        For i = 1 To MAX_MORALS
            ClearMoral(i)
        Next
    End Sub

    Friend Sub StreamMoral(moralNum As Integer)
        If moralnum > 0 and Type.Moral(moralNum).Name = "" Or GameState.Moral_Loaded(moralNum) = 0 Then
            GameState.Moral_Loaded(moralNum) = 1
            SendRequestMoral(moralNum)
        End If
    End Sub

#End Region

#Region "Incoming Packets"

#End Region
End Module

File: Source/Client/Game/Objects/Moral.cs
using Core;
using Microsoft.VisualBasic.CompilerServices;

namespace Client
{

    public class Moral
    {
        #region Database

        public static void ClearMoral(int index)
        {
            Data.Moral[index] = default;

            Data.Moral[index].Name = "";
            GameState.Moral_Loaded[index] = 0;
        }

        public static void ClearMorals()
        {
            int i;

            Data.Moral = new Core.Type.Moral[(Constant.MAX_MORALS)];

            for (i = 0; i < Constant.MAX_MORALS; i++)
                ClearMoral(i);
        }

        public static void StreamMoral(int moralNum)
        {
            if (moralNum >= 0 & string.IsNullOrEmpty(Data.Moral[moralNum].Name) && GameState.Moral_Loaded[moralNum] == 0)
            {
                GameState.Moral_Loaded[moralNum] = 1;
                NetworkSend.SendRequestMoral(moralNum);
            }
        }

        #endregion

        #region Incoming Packets

        #endregion
    }
}